internal/storage/references/registry_reference.go

```diff
@@ -36,6 +36,24 @@ func RegistryImageReferenceFromRaw(rawNamed reference.Named) RegistryImageRefere
 	return RegistryImageReference{privateNamed: rawNamed}
 }
 
+// RegistryImageReferenceFromRawSerialized is an internal constructor of a RegistryImageReference, that also
+// strips the tag from ambiguous image references that have a digest as well (e.g.  `image:tag@sha256:123...`).
+// Such image references are supported by docker but, due to their ambiguity, explicitly not by containers/image.
+//
+// This should only be called from internal/storage.
+func RegistryImageReferenceFromRawSerialized(rawNamed reference.Named) (RegistryImageReference, error) {
```

Comment:
not crazy about this name, @mtrmac do you have any suggestions?

Reply:
See the other comment for a bit higher-level view…

… which still leaves open the naming of this canonicalization. My starting point is `canonicalize{TagDigest,DockerLike}` somewhere in the name, but that might get awfully long.